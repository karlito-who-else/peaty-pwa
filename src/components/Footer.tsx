import React from "react";
import { makeStyles, Theme, createStyles } from "@material-ui/core/styles";
import Container from "@material-ui/core/Container";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemAvatar from "@material-ui/core/ListItemAvatar";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemSecondaryAction from "@material-ui/core/ListItemSecondaryAction";
import ListItemText from "@material-ui/core/ListItemText";
import Avatar from "@material-ui/core/Avatar";
import IconButton from "@material-ui/core/IconButton";
import Grid from "@material-ui/core/Grid";
import Typography from "@material-ui/core/Typography";
import FolderIcon from "@material-ui/icons/Folder";
import DeleteIcon from "@material-ui/icons/Delete";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      backgroundColor: theme.palette.background.paper,
      flexGrow: 1
    },
    container: {},
    demo: {
      backgroundColor: theme.palette.background.paper
    },
    title: {
      margin: theme.spacing(4, 0, 2)
    }
  })
);

function generate(element: React.ReactElement): ReactElement[] {
  return [0, 1, 2].map(value =>
    React.cloneElement(element, {
      key: value
    })
  );
}

export const InteractiveList = (): ReactElement => {
  const classes = useStyles();

  return (
    <footer className={classes.root}>
      <Container className={classes.container} component="div" fixed>
        <Grid container spacing={2}>
          <Grid item xs={12} md={3}>
            <Typography variant="h6" className={classes.title}>
              Text only
            </Typography>
            <div className={classes.demo}>
              <List>
                {generate(
                  <ListItem>
                    <ListItemText primary="Single-line item" />
                  </ListItem>
                )}
              </List>
            </div>
          </Grid>
          <Grid item xs={12} md={3}>
            <Typography variant="h6" className={classes.title}>
              Icon with text
            </Typography>
            <div className={classes.demo}>
              <List>
                {generate(
                  <ListItem>
                    <ListItemIcon>
                      <FolderIcon />
                    </ListItemIcon>
                    <ListItemText primary="Single-line item" />
                  </ListItem>
                )}
              </List>
            </div>
          </Grid>
          <Grid item xs={12} md={3}>
            <Typography variant="h6" className={classes.title}>
              Avatar with text
            </Typography>
            <div className={classes.demo}>
              <List>
                {generate(
                  <ListItem>
                    <ListItemAvatar>
                      <Avatar>
                        <FolderIcon />
                      </Avatar>
                    </ListItemAvatar>
                    <ListItemText primary="Single-line item" />
                  </ListItem>
                )}
              </List>
            </div>
          </Grid>
          <Grid item xs={12} md={3}>
            <Typography variant="h6" className={classes.title}>
              Avatar with text and icon
            </Typography>
            <div className={classes.demo}>
              <List>
                {generate(
                  <ListItem>
                    <ListItemAvatar>
                      <Avatar>
                        <FolderIcon />
                      </Avatar>
                    </ListItemAvatar>
                    <ListItemText primary="Single-line item" />
                    <ListItemSecondaryAction>
                      <IconButton edge="end" aria-label="delete">
                        <DeleteIcon />
                      </IconButton>
                    </ListItemSecondaryAction>
                  </ListItem>
                )}
              </List>
            </div>
          </Grid>
        </Grid>
      </Container>
    </footer>
  );
};

export default InteractiveList;
