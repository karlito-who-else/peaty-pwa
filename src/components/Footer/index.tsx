import React from "react";

import {
  Avatar,
  IconButton,
  List,
  ListItem,
  ListItemAvatar,
  ListItemIcon,
  ListItemSecondaryAction,
  ListItemText,
  Theme,
  Typography
} from "@material-ui/core";
import { createStyles, makeStyles } from "@material-ui/core/styles";
import { Delete as DeleteIcon, Folder as FolderIcon } from "@material-ui/icons";

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      backgroundColor: theme.palette.background.paper,
      display: "grid",
      gap: `${theme.spacing(2)}px`,
      gridArea: "footer",
      gridAutoRows: `minmax(100px, auto)`,
      gridTemplateColumns: `repeat(2, 1fr)`,
      [theme.breakpoints.up("lg")]: {
        gridTemplateColumns: `repeat(4, 1fr)`
      },
      padding: theme.spacing(2)
    },
    title: {
      margin: theme.spacing(4, 0, 2)
    }
  })
);

function generate(element: React.ReactElement): ReactElement[] {
  return [0, 1, 2].map(value =>
    React.cloneElement(element, {
      key: value
    })
  );
}

export const Footer = (): ReactElement => {
  const classes = useStyles();

  return (
    <footer className={classes.root}>
      <div>
        <Typography variant="h6" className={classes.title}>
          Text only
        </Typography>
        <List>
          {generate(
            <ListItem>
              <ListItemText primary="Single-line item" />
            </ListItem>
          )}
        </List>
      </div>
      <div>
        <Typography variant="h6" className={classes.title}>
          Icon with text
        </Typography>
        <List>
          {generate(
            <ListItem>
              <ListItemIcon>
                <FolderIcon />
              </ListItemIcon>
              <ListItemText primary="Single-line item" />
            </ListItem>
          )}
        </List>
      </div>
      <div>
        <Typography variant="h6" className={classes.title}>
          Avatar with text
        </Typography>
        <List>
          {generate(
            <ListItem>
              <ListItemAvatar>
                <Avatar>
                  <FolderIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary="Single-line item" />
            </ListItem>
          )}
        </List>
      </div>
      <div>
        <Typography variant="h6" className={classes.title}>
          Avatar with text and icon
        </Typography>
        <List>
          {generate(
            <ListItem>
              <ListItemAvatar>
                <Avatar>
                  <FolderIcon />
                </Avatar>
              </ListItemAvatar>
              <ListItemText primary="Single-line item" />
              <ListItemSecondaryAction>
                <IconButton edge="end" aria-label="delete">
                  <DeleteIcon />
                </IconButton>
              </ListItemSecondaryAction>
            </ListItem>
          )}
        </List>
      </div>
    </footer>
  );
};

export default Footer;
