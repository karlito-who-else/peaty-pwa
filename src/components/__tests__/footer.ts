import React from "react";
import renderer from "react-test-renderer";

import Footer from "../Footer";

describe("Footer", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Footer showTitles={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
