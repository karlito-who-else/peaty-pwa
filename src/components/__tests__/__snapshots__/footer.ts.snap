// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer
  className="footer has-background-black has-text-white"
>
  <div
    className="content has-text-centered"
  >
    <img
      alt="Kaldi"
      src="test-file-stub"
      style={
        Object {
          "height": "10em",
          "width": "14em",
        }
      }
    />
  </div>
  <div
    className="content has-text-centered has-background-black has-text-white"
  >
    <div
      className="container has-background-black has-text-white"
    >
      <div
        className="columns"
      >
        <div
          className="column is-4"
        >
          <section
            className="menu"
          >
            <ul
              className="menu-list"
            >
              <li>
                <a
                  className="navbar-item"
                  href="/"
                >
                  Home
                </a>
              </li>
              <li>
                <a
                  className="navbar-item"
                  href="/about"
                >
                  About
                </a>
              </li>
              <li>
                <a
                  className="navbar-item"
                  href="/jobs"
                >
                  Jobs
                </a>
              </li>
              <li>
                <a
                  className="navbar-item"
                  href="/products"
                >
                  Products
                </a>
              </li>
              <li>
                <a
                  className="navbar-item"
                  href="/contact/examples"
                >
                  Form Examples
                </a>
              </li>
              <li>
                <a
                  className="navbar-item"
                  href="/admin/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Admin
                </a>
              </li>
            </ul>
          </section>
        </div>
        <div
          className="column is-4"
        >
          <section>
            <ul
              className="menu-list"
            >
              <li>
                <a
                  className="navbar-item"
                  href="/blog"
                >
                  Latest Stories
                </a>
              </li>
              <li>
                <a
                  className="navbar-item"
                  href="/contact"
                >
                  Contact
                </a>
              </li>
            </ul>
          </section>
        </div>
        <div
          className="column is-4 social"
        >
          <a
            href="https://facebook.com"
            title="facebook"
          >
            <img
              alt="Facebook"
              src="test-file-stub"
              style={
                Object {
                  "height": "1em",
                  "width": "1em",
                }
              }
            />
          </a>
          <a
            href="https://twitter.com"
            title="twitter"
          >
            <img
              alt="Twitter"
              className="fas fa-lg"
              src="test-file-stub"
              style={
                Object {
                  "height": "1em",
                  "width": "1em",
                }
              }
            />
          </a>
          <a
            href="https://instagram.com"
            title="instagram"
          >
            <img
              alt="Instagram"
              src="test-file-stub"
              style={
                Object {
                  "height": "1em",
                  "width": "1em",
                }
              }
            />
          </a>
          <a
            href="https://vimeo.com"
            title="vimeo"
          >
            <img
              alt="Vimeo"
              src="test-file-stub"
              style={
                Object {
                  "height": "1em",
                  "width": "1em",
                }
              }
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
